eng/common/scripts/modules/ChangeLog-Operations.psm1

```diff
@@ -0,0 +1,110 @@
+# Common Changelog Operations
+
+$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+.*(?<version>\b\d+\.\d+\.\d+([^0-9\s][^\s:]+)?)(\s(?<releaseStatus>\(Unreleased\)|\(\d{4}-\d{2}-\d{2}\)))?)"
+
+# given a CHANGELOG.md file, extract the relevant info we need to decorate a release
+function Get-ReleaseNotes {
+    param (
+        [Parameter(Mandatory = $true)]
+        [String]$ChangeLogLocation
+    )
+    
+    $ErrorActionPreference = 'Stop'
+
+    $releaseNotes = @{}
+    if ($ChangeLogLocation.Length -eq 0)
+    {
+        return $releaseNotes
+    }
+    
+    try 
+    {
+        $contents = Get-Content $ChangeLogLocation
+        # walk the document, finding where the version specifiers are and creating lists
+        $releaseNotesEntry = $null
+        foreach($line in $contents){
+            if ($line -match $RELEASE_TITLE_REGEX)
+            {
+               $releaseNotesEntry = [pscustomobject]@{ 
+                  ReleaseVersion = $matches["version"]
+                  ReleaseStatus = $matches["releaseStatus"]
+                  ReleaseTitle = $line
+                  ReleaseContent = @() # Release content without the version title
+               }
+               $releaseNotes[$releaseNotesEntry.ReleaseVersion] = $releaseNotesEntry
+            }
+            else
+            {
+               if ($releaseNotesEntry) {
+                  $releaseNotesEntry.ReleaseContent += $line
+               }
+            }
+        }
+    }
+    catch
+    {
+        Write-Host "Error parsing $ChangeLogLocation."
+        Write-Host $_.Exception.Message
+    }
+    return $releaseNotes
+}
+
+function Get-ReleaseNote {
```

Comment:
Perhaps we have a Get-ReleaseNoteEntry and Get-ReleaseNoteContent that returns the title and the content joined by newline.